crypto/openssl_common.go

```diff
@@ -94,5 +104,13 @@ func (sk *opensslPrivateKey) Raw() ([]byte, error) {
 
 // Equals checks whether this key is equal to another
 func (sk *opensslPrivateKey) Equals(k Key) bool {
-	return KeyEqual(sk, k)
+	a, err := sk.Raw()
+	if err != nil {
+		return false
+	}
+	b, err := k.Raw()
+	if err != nil {
+		return false
+	}
```

Comment:
helper?

Reply:
meh

Reply:
Well, I fixed it anyways... (mostly so I could consistently fix https://github.com/libp2p/go-libp2p-core/pull/28#discussion_r298548461)

Reply:
thanks